src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -421,21 +418,19 @@ private ActionResult ExceptionResponse(Exception exception, string message)
         {
             _logger.LogError(exception, message);
 
-            if (exception is PlatformHttpException)
+            if (exception is PlatformHttpException phe)
             {
-                PlatformHttpException phe = exception as PlatformHttpException;
                 return StatusCode((int)phe.Response.StatusCode, phe.Message);
             }
-            else if (exception is ServiceException)
+            else if (exception is ServiceException se)
             {
-                ServiceException se = exception as ServiceException;
                 return StatusCode((int)se.StatusCode, se.Message);
             }
 
             return StatusCode(500, $"{message}");
         }
 
-        private async Task<ActionResult> CreateBinaryData(Instance instanceBefore, string dataType, string contentType, string filename, Stream fileStream)
+        private async Task<ActionResult> CreateBinaryData(Instance instanceBefore, string dataType, string contentType, string? filename, Stream fileStream)
```

Comment:
Nice, I only traced it to being read from a header I think is optional, so I figured it might be null. We should probably store the file with a filename anyway, so a better solution to the nullability problem might be to throw or use `Guid.NewGuid()`